Controllers/UserAuthenticationController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+namespace Controllers;
+use models\User;
+
+class UserAuthenticationController extends BasicController{
+
+	public function __construct() {
+		parent::__construct();
+		$this->title = "User Authentication";
+	}
+
+	/**
+	 * Callback for /register route.
+	 */
+	public function registerPageAction() {
+		$message = render_messages(get_messages());
+		$this->content = $this->render('views/forms/register.tpl.php', array('message' => $message));
+		$this->renderLayout('/views/layouts/basic.tpl.php');
+	}
+
+	/**
+	 * Registers user if credentials are valid.
+	 */
+	public function registerPost(){
+		if (isset($_POST['email']) && isset($_POST['password']) && isset($_POST['confirm-password'])){
+			$email = $_POST['email'];
+			$password = $_POST['password'];
+			$passwordConfirmation = $_POST['confirm-password'];
+			$db = db_get_connection();
```

Comment:
try to include this line directly in BasicModel contructor, instead of getting $db as argument and make sure you create constructor function for User model, that is calling parent constructor.